spec/lib/rmt/scc_spec.rb

```diff
@@ -131,4 +141,69 @@ def scc
       expect { other_repo_without_token.reload }.not_to raise_error
     end
   end
+
+  describe '#export' do
+    let(:path) { '/tmp/usb' }
+
+    context 'without SCC credentials' do
+      before do
+        allow(Settings).to receive(:scc).and_return OpenStruct.new
+      end
+
+      it 'raises an error' do
+        expect { described_class.new.export(path) }.to raise_error RMT::SCC::CredentialsError, 'SCC credentials not set.'
+      end
+    end
+
+    context 'with SCC credentials' do
+      before do
+        allow(Settings).to receive(:scc).and_return(OpenStruct.new(username: 'me', password: 'groot'))
+      end
+
+      it 'makes sure path exists' do
+        expect(File.directory?(path)).to be true
+      end
+
+      %w[orders products repositories subscriptions].each do |data|
+        it "writes #{data} file to path" do
+          FakeFS.with_fresh do
+            described_class.new.export(path)
+            expect(File.exist?(File.join(path, "organizations_#{data}.json")))
+          end
+        end
+      end
+    end
+  end
+
+  describe '#import' do
+    let(:path) { '/tmp/usb' }
+
+    context 'with bad path or missing files' do
+      it 'raises an error before it touches the database' do
+        FakeFS.with_fresh do
+          FakeFS::FileSystem.clone(Rails.root.join('app', 'models')) # needed for Rails auto-loading to still work
+          expect(ApplicationRecord).not_to receive(:delete_all)
+          expect(ApplicationRecord).not_to receive(:update)
+          expect(ApplicationRecord).not_to receive(:new)
+          expect { described_class.new.import(path) }.to raise_error RMT::SCC::DataFilesError
+        end
+      end
+    end
+
+    context 'with good path and files' do
+      before do
+        FakeFS.with_fresh do
+          FakeFS::FileSystem.clone(Rails.root.join('app', 'models')) # needed for Rails auto-loading to still work
+          FileUtils.mkdir_p(path)
+          File.write(File.join(path, 'organizations_products.json'), products.to_json)
+          File.write(File.join(path, 'organizations_subscriptions.json'), subscriptions.to_json)
+          File.write(File.join(path, 'organizations_repositories.json'), all_repositories.to_json)
+
+          described_class.new.import(path)
+        end
+      end
+
+      include_examples 'saves in database'
```

Comment:
Those shared examples are somehow order dependent/have some context from the previous run.
If you run the whole spec file -- it works, however when running just the `decribe '#import'` block they fail.
Hmm, maybe we should enable random execution order for all the specs?

Reply:
Hmm not good. Is this still the case?